pkg/localstorev2/internal/upload/uploadstore_test.go

```diff
@@ -297,59 +462,132 @@ func TestChunkPutter(t *testing.T) {
+				if err != nil {
+					t.Fatalf("Report(...): unexpected error: %v", err)
 				}
 			})
 
 			t.Run("verify internal state", func(t *testing.T) {
-				has, err := ts.Store().Has(&upload.TagIDAddressItem{
-					TagID:   tagID,
+				ti := &upload.TagItem{
+					TagID: tagID,
+				}
+				err := ts.Store().Get(ti)
+				if err != nil {
+					t.Fatalf("Get(...): unexpected error: %v", err)
+				}
+				count := uint64(idx + 1)
+				exp := &upload.TagItem{
+					TagID:     tagID,
+					StartedAt: now().Unix(),
+					Sent:      count,
+					Synced:    count,
+					Stored:    count,
+				}
+
+				if diff := cmp.Diff(ti, exp); diff != "" {
+					t.Fatalf("Get(...): unexpected TagItem (-want +have):\n%s", diff)
+				}
+
+				ui := &upload.UploadItem{
 					Address: chunk.Address(),
-				})
+					BatchID: chunk.Stamp().BatchID(),
+				}
+				err = ts.Store().Get(ui)
 				if err != nil {
-					t.Fatalf("Has(...): unexpected error: %v", err)
+					t.Fatalf("Get(...): unexpected error: %v", err)
 				}
-				if !has {
-					t.Fatal("Has(...): item not found")
+				uexp := &upload.UploadItem{
+					Address:  chunk.Address(),
+					BatchID:  chunk.Stamp().BatchID(),
+					TagID:    tagID,
+					Uploaded: now().Unix(),
+					Synced:   now().Unix(),
 				}
 
-				has, err = ts.Store().Has(&upload.PushItem{
+				if diff := cmp.Diff(ui, uexp); diff != "" {
+					t.Fatalf("Get(...): unexpected UploadItem (-want +have):\n%s", diff)
+				}
+
+				pi := &upload.PushItem{
 					Timestamp: now().Unix(),
 					Address:   chunk.Address(),
-					TagID:     tagID,
-				})
+					BatchID:   chunk.Stamp().BatchID(),
+				}
+				has, err := ts.Store().Has(pi)
 				if err != nil {
 					t.Fatalf("Has(...): unexpected error: %v", err)
 				}
-				if !has {
-					t.Fatalf("Has(...): item not found")
+				if has {
+					t.Fatalf("Has(...): expected to not be found: %s", pi)
 				}
 
-				have, err := ts.ChunkStore().Get(context.TODO(), chunk.Address())
+				have, err := ts.ChunkStore().Has(context.TODO(), chunk.Address())
 				if err != nil {
 					t.Fatalf("Get(...): unexpected error: %v", err)
 				}
-				if want := chunk; !want.Equal(have) {
-					t.Fatalf("Get(...): chunk missmatch:\nwant: %x\nhave: %x", want, have)
+				if have {
+					t.Fatalf("Get(...): chunk expected to not be found: %s", chunk.Address())
 				}
 			})
 		})
 	}
+
+	t.Run("close with reference", func(t *testing.T) {
+		addr := swarmtesting.RandomAddress()
+
+		err := putter.Close(addr)
+		if err != nil {
+			t.Fatalf("Close(...): unexpected error %v", err)
+		}
+
+		ti := &upload.TagItem{TagID: tagID}
+		err = ts.Store().Get(ti)
+		if err != nil {
+			t.Fatalf("Get(...): unexpected error %v", err)
+		}
+
+		exp := &upload.TagItem{
+			TagID:     tagID,
+			Split:     10,
+			Seen:      0,
+			Stored:    10,
+			Sent:      10,
+			Synced:    10,
+			StartedAt: now().Unix(),
+			Address:   addr,
+		}
+		if diff := cmp.Diff(ti, exp); diff != "" {
```

Comment:
The first argument should be `exp` (a.k.a `want`) and the second should be `ti` (a.k.a `have`).
Also, check this in other places where you use `cmp.Diff`.